x/twap/logic_test.go

```diff
@@ -42,28 +43,43 @@ func (s *TestSuite) TestGetSpotPrices() {
 		expectedSp1           sdk.Dec
 		expectedLatestErrTime time.Time
 	}{
-		"zero sp": {
+		"use last spot price on err": {
 			poolID:                poolID,
 			prevErrTime:           currTime,
+			blockTime:             s.Ctx.BlockTime(),
 			mockSp0:               sdk.ZeroDec(),
 			mockSp1:               sdk.ZeroDec(),
 			mockSp0Err:            fmt.Errorf("foo"),
-			expectedSp0:           sdk.ZeroDec(),
+			expectedSp0:           sdk.OneDec(),
```

Comment:
why does expectedSp0 become oneDec? (both mocksp0 and mocksp1) being zero dec, I was expecting expectedSp1 to be zero dec as well?

Reply:
In this case, we got err when getting sp0. `blockTime` coincides with pool creation time so we have a record with sp0 = sp1 = 1. 
expectedSp0 take the old record sp0.

Reply:
hrrm but the last sp1 before the update was zero as well right? Shouldnt expectedSp1 stay zero as well?

Reply:
With sp1 we don't get any error so I think it's updated normally (is zero as mock)